File: app/src/main/java/com/firelord/slidenews/domain/usecase/GetSavedNewsUseCase.kt
package com.firelord.slidenews.domain.usecase

import com.firelord.slidenews.data.model.Article
import com.firelord.slidenews.domain.repository.NewsRepository
import kotlinx.coroutines.flow.Flow

class GetSavedNewsUseCase(private val newsRepository: NewsRepository) {
    fun execute():Flow<List<Article>>{
        return newsRepository.getSavedNews()
    }
}

File: Slide News/domain/usecase/news/db/GetSavedNewsUseCase.swift
//
//  GetSavedNewsUseCase.swift
//  Slide News
//
//  Created by admin on 26/04/24.
//

import Foundation

final class GetSavedNewsUseCase {
    let newsRepository: NewsRepository
    
    init(newsRepository: NewsRepository) {
        self.newsRepository = newsRepository
    }
    
    func execute() async throws -> [Article] {
        return try await newsRepository.getSavedNews()
    }
}
